File: unity/Parallaxer/Assets/Scripts/Parallaxer.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Parallaxer : MonoBehaviour {

    public float speed;

    private Parallaxer dupeLayer;
    private bool isDupe = false;
    private float spriteWidth;
    private float startX, endX;
    private Vector2 delta = Vector2.zero;

    private void Start() {
        if (gameObject.name.Split('(')[0] != transform.parent.name.Split('(')[0]) {
            dupeLayer = GameObject.Instantiate(gameObject).GetComponent<Parallaxer>();
            dupeLayer.isDupe = true;
            dupeLayer.transform.SetParent(transform);

            SpriteRenderer ren = GetComponent<SpriteRenderer>();
            spriteWidth = ren.sprite.rect.width / 100f; // half pixel width / 100

            transform.Translate(new Vector3(spriteWidth, 0f, 0f));
            startX = transform.position.x;
            endX = startX - spriteWidth;

            dupeLayer.startX = startX - spriteWidth;
            dupeLayer.transform.position = new Vector2(dupeLayer.startX, transform.position.y);
        }
    }

    private void Update() {
        if (!isDupe) {
            delta = Vector2.left * speed * Time.deltaTime;

            transform.Translate(delta);

            if (transform.position.x < endX) {
				transform.position = new Vector2(startX, transform.position.y);
            } else if (transform.position.x > startX) {
                transform.position = new Vector2(endX, transform.position.y);
            }
        }
    }

}

File: p5js/ParallaxExample/js/parallaxer.js
"use strict";

class Parallaxer {

	constructor(_img, _speed, _position) {
		this.img = _img;
		this.speed = _speed;
		this.position = _position;
		
		this.spriteWidth;
		this.startX;
		this.endX;
		this.position;
		this.position2;

		this.init();
	}
	 
	init() {
		this.spriteWidth = this.img.width;
		this.startX = this.position.x;
		this.endX = this.startX - this.spriteWidth;
		this.position2 = createVector(this.startX + this.spriteWidth, 0);
	}
 
	update() {
		this.position.x -= this.speed;
		
		if (this.position.x < this.endX) {
			this.position = createVector(this.startX, this.position.y);
		} else if (this.position.x > this.startX) {
			this.position = createVector(this.endX, this.position.y);
		}
	}
	
	draw() {
		push();
		translate(this.position.x, this.position.y);
		image(this.img, 0, 0);
		image(this.img, this.position2.x, this.position2.y);
		pop();
	}
	
	run() {
		this.update();
		this.draw();
	}

}
